controllers/metal3.io/host_state_machine_test.go

```diff
@@ -146,6 +146,278 @@ func TestProvisioningCapacity(t *testing.T) {
+		},
+		{
+			Scenario: "PreparingHost",
+			Host:     host(metal3v1alpha1.StatePreparing).build(),
+			Annotations: map[string]string{
+				metal3v1alpha1.DetachedAnnotation: "true",
+			},
+			ExpectedDetach:            false,
+			ExpectedDirty:             true,
+			ExpectedOperationalStatus: metal3v1alpha1.OperationalStatusOK,
+			ExpectedState:             metal3v1alpha1.StateReady,
+		},
+		{
+			Scenario: "ReadyHost",
+			Host:     host(metal3v1alpha1.StateReady).build(),
+			Annotations: map[string]string{
+				metal3v1alpha1.DetachedAnnotation: "true",
+			},
+			ExpectedDetach:            false,
+			ExpectedDirty:             true,
+			ExpectedOperationalStatus: metal3v1alpha1.OperationalStatusOK,
+			ExpectedState:             metal3v1alpha1.StatePreparing,
+		},
+		{
+			Scenario: "ProvisioningHost",
+			Host:     host(metal3v1alpha1.StateProvisioning).build(),
+			Annotations: map[string]string{
+				metal3v1alpha1.DetachedAnnotation: "true",
+			},
+			ExpectedDetach:            false,
+			ExpectedDirty:             true,
+			ExpectedOperationalStatus: metal3v1alpha1.OperationalStatusOK,
+			ExpectedState:             metal3v1alpha1.StateProvisioned,
+		},
+		{
+			Scenario: "DeprovisioningHost",
+			Host:     host(metal3v1alpha1.StateDeprovisioning).build(),
+			Annotations: map[string]string{
+				metal3v1alpha1.DetachedAnnotation: "true",
+			},
+			ExpectedDetach:            false,
+			ExpectedDirty:             true,
+			ExpectedOperationalStatus: metal3v1alpha1.OperationalStatusOK,
+			ExpectedState:             metal3v1alpha1.StateReady,
+		},
+		{
+			Scenario: "DeletingHost",
+			Host:     host(metal3v1alpha1.StateDeleting).setDeletion().build(),
+			Annotations: map[string]string{
+				metal3v1alpha1.DetachedAnnotation: "true",
+			},
+			ExpectedDetach:            false,
+			ExpectedDirty:             false,
+			ExpectedOperationalStatus: metal3v1alpha1.OperationalStatusOK,
+			ExpectedState:             metal3v1alpha1.StateDeleting,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.Scenario, func(t *testing.T) {
+			tc.Host.Annotations = tc.Annotations
+			prov := newMockProvisioner()
+			if tc.ExpectedError {
+				prov.setNextError("Detach", "some error")
+			}
+			hsm := newHostStateMachine(tc.Host, &BareMetalHostReconciler{}, prov, true)
+			info := makeDefaultReconcileInfo(tc.Host)
+			result := hsm.ReconcileState(info)
+
+			assert.Equal(t, tc.ExpectedDetach, prov.isDetached, "ExpectedDetach mismatch")
+			assert.Equal(t, tc.ExpectedDirty, result.Dirty(), "ExpectedDirty mismatch")
+			assert.Equal(t, tc.ExpectedOperationalStatus, info.host.OperationalStatus())
+			assert.Equal(t, tc.ExpectedState, info.host.Status.Provisioning.State)
+		})
+	}
+}
+
+func TestDetachError(t *testing.T) {
+	testCases := []struct {
+		Scenario                  string
+		Host                      *metal3v1alpha1.BareMetalHost
+		Annotations               map[string]string
```

Comment:
Looks unused

Reply:
Thanks, yes copy/paste error - removed in the latest version